src/compiler.ts

```diff
@@ -8700,27 +8714,94 @@ export class Compiler extends DiagnosticEmitter {
       }
       let fieldInstance = <Field>member;
       let fieldType = fieldInstance.type;
-      exprs[i + 1] = this.module.store( // TODO: handle setters as well
-        fieldType.byteSize,
-        this.module.local_get(tempLocal.index, this.options.nativeSizeType),
-        this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT),
-        fieldType.toNativeType(),
-        fieldInstance.memoryOffset
-      );
+
+      let expr = this.compileExpression(values[i], fieldType, Constraints.CONV_IMPLICIT | Constraints.WILL_RETAIN);
+      if (isManaged && fieldType.isManaged && !this.skippedAutoreleases.has(expr)) {
+        expr = this.makeRetain(expr);
+      }
+      exprs.push(this.module.store( // TODO: handle setters as well
+       fieldType.byteSize,
+       this.module.local_get(tempLocal.index, this.options.nativeSizeType),
+       expr,
+       fieldType.toNativeType(),
+       fieldInstance.memoryOffset
+      ));
+
+      // This member is no longer omitted, so delete from our omitted fields
+      omittedClassFieldMembers.delete(member.name);
     }
     this.currentType = classReference.type.nonNullableType;
     if (hasErrors) return module.unreachable();
 
+    // Iterate through the remaining omittedClassFieldMembers.
+    if (members) {
+      for(let _values = Set_values(omittedClassFieldMembers), i = 0, v = _values.length; i < v; ++i) {
+        let omittedMemberKey = _values[i];
+        let member = assert(members.get(omittedMemberKey));
+        
+        let fieldInstance = <Field>member;
+        let fieldType = fieldInstance.type;
+
+        if (fieldType.is(TypeFlags.REFERENCE) && fieldType.classReference !== null) {
+          // TODO: Check if it is a class, with a default value (constructor with no params).
+          // TODO: Check if it can be null, and set to null
+
+          // Otherwise, error
+          this.error(
+            DiagnosticCode.Property_0_is_missing_in_type_1_but_required_in_type_2,
+            expression.range, "<object>", classReference.toString()
+          );
+          hasErrors = true;
+          continue;
```

Comment:
Reminds me that you asked for this for your use case, and this can be supported by only erroring if the field's type is not nullable, and otherwise letting it store a `0` below:

```suggestion
          if (!fieldType.is(TypeFlags.NULLABLE)) {
            this.error(
              DiagnosticCode.Property_0_is_missing_in_type_1_but_required_in_type_2,
              expression.range, "<object>", classReference.toString()
            );
            hasErrors = true;
            continue;
          }
```

Reply:
Oooooo nice! Thank you for suggesting that quick fix for support! :smile: :+1: